mos-platform/atari2600-3e/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+platform(atari2600-3e COMPLETE PARENT atari2600)
```

Comment:
I'll do a more thorough review in a bit, but I wanted to get this comment out early to save time. It's generally not desirable to have one complete target as a parent of another; targets should be open for extension, but nothing in a target should override or replace anything in its parent. Accordingly, all of the contents of a parent's e.g. libc are silently merged into a derived targets library of the same name. This is generally not what you want for situations like this.

Instead, both the "basic" `atari2600` and `atari2600-3e` targets should be made to derive from an incomplete `atari2600-common` target that has the functionality that is truly common between them.

I'm also not very familiar with the 2600, so I'll leave this to you: we should only call one of these targets `atari2600` if it's overwhelmingly the configuration that folks in the community think of when developing for it. (I'm guessing it's the most common commercial system cartridge size?)

Reply:
I thought I could get away with a shared crt0, but I think it's a good idea to make a common parent. The 4K mapper is closest to NES NROM in that it's basically no mapper. But although it's the most straightforward, it's not the most useful for new development, so I think explicitly calling it atari2600-4k is good.

I was having some intermittent problems with the init code going away unless I added an optimization flag (?) but that seems fixed, maybe it was a CMake thing. Also not quite sure on weak functions, they were sometimes getting zeroed out when I tried to invoke them.

Since 2600 carts have no header, emulators use a bunch of heuristics to determine the mapper, and it doesn't always work right. 4K is fine, but current emulators will probably have to be told the 3E example ROM is 3E. Maybe there can be a "autodetection helper library" to smooth things out.